handlers.go

```diff
@@ -4208,11 +4209,40 @@ func (s *Server) handleColfsListDir(c echo.Context, u *User) error {
 			return err
 		}
 
+		// Build a new local DAGService to query for the CID if our ref and see if it's a directory
+		bserv := blockservice.New(s.Node.Blockstore, offline.Exchange(s.Node.Blockstore))
+		dserv := merkledag.NewDAGService(bserv)
+		ctx := c.Request().Context()
+		nd, err := dserv.Get(ctx, r.Cid.CID)
```

Comment:
hrm, is it possible that trying to look up the CID from the local blockstore fails if pinning in still in progress in `ContentManager`?

if so, I think we may just want to either exclude the `CollectionRef` in the results, or revert to previous behavior instead. as it stands it looks like any CID lookup failure in the local blockstore will make the collection fs list fail altogether

Reply:
Yeah, this is an interesting edge case. If content is offloaded then listing will fail. Maybe we should encode some additional information in the collectionRef?

We could also include an extra field in the collectionListResponse to indicate the content is not current accessible on estuary? call it 'Active' or not

Reply:
@gmelodie that sounds totally reasonable to me!

Reply:
the pin status endpoint gets you that info, see `handleGetPin`

Reply:
and +1 to your suggestion there, seems reasonable to not list it if its in progress